src/commands/AdventOfCodeCommand.ts

```diff
@@ -0,0 +1,134 @@
+import { ColorResolvable, MessageEmbed, CommandInteraction } from "discord.js";
+import AdventOfCodeService from "../services/AdventOfCodeService";
+import { AOCMember } from "../interfaces/AdventOfCode";
+import getConfigValue from "../utils/getConfigValue";
+import GenericObject from "../interfaces/GenericObject";
+import {Discord, Slash, SlashOption} from "discordx";
+
+@Discord()
+class AdventOfCodeCommand {
+	@Slash("aoc", {description: "Advent Of Code"})
+	async onInteract(
+		@SlashOption("year", {type: "NUMBER", minValue: 2015, required: false}) year: number,
+		@SlashOption("name", {type: "STRING", required: false}) name: string,
+			interaction: CommandInteraction): Promise<void> {
+		const currentAOCYear = this.getYear();
+		const adventOfCodeService = AdventOfCodeService.getInstance();
+		const embed = new MessageEmbed();
+		let yearToQuery = currentAOCYear;
+
+		if (!!year && year <= yearToQuery) {
+			yearToQuery = year;
+		} else if (!!year) {
+			await interaction.reply({embeds: [this.errorEmbed(`Year requested not available.\nPlease query a year between 2015 and ${yearToQuery}`)], ephemeral: true});
+			return;
+		}
+
+		const link = `https://adventofcode.com/${yearToQuery}/leaderboard/private/view/${getConfigValue<string>("ADVENT_OF_CODE_LEADERBOARD")}`;
+		const description = `Leaderboard ID (${currentAOCYear}): \`${getConfigValue<string>("ADVENT_OF_CODE_INVITE")}\`\n\n[View Leaderboard (${yearToQuery})](${link})`;
```

Comment:
We could use a Button component for the link now.

Reply:
Added! The description is now simply the leaderboard ID, with the button linking to the leaderboard.